From 6c7346b579898f9017ec98e62bfc6454a427ab05 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Zbigniew=20J=C4=99drzejewski-Szmek?= <zbyszek@in.waw.pl>
Date: Thu, 9 Feb 2023 22:37:01 +0100
Subject: [PATCH] Revert "logind: implement Type=notify-reload protocol
 properly"

This reverts commit 5d71e463f49518c7702467f6145484afa31bf8ba.

It turns out that this commit caused a noticable change in behaviour for
'systemctl isolate graphical.target' in Fedora, as found by git bisect.
Reverting on top of current git also restores behaviour from v252. I don't have
time to analyze this right now, so this is a quick revert to unblock Fedora
and possibly allow us to release v253 in case a full solution is harder.

Fixes #26364.
---
 src/login/logind.c              | 6 ------
 units/systemd-logind.service.in | 1 -
 2 files changed, 7 deletions(-)

diff --git a/src/login/logind.c b/src/login/logind.c
index 1feacd3601..def2f5a442 100644
--- a/src/login/logind.c
+++ b/src/login/logind.c
@@ -1020,11 +1020,6 @@ static int manager_dispatch_reload_signal(sd_event_source *s, const struct signa
         Manager *m = userdata;
         int r;
 
-        (void) sd_notifyf(/* unset= */ false,
-                          "RELOADING=1\n"
-                          "STATUS=Reloading configuration...\n"
-                          "MONOTONIC_USEC=" USEC_FMT, now(CLOCK_MONOTONIC));
-
         manager_reset_config(m);
         r = manager_parse_config_file(m);
         if (r < 0)
@@ -1032,7 +1027,6 @@ static int manager_dispatch_reload_signal(sd_event_source *s, const struct signa
         else
                 log_info("Config file reloaded.");
 
-        (void) sd_notify(/* unset= */ false, NOTIFY_READY);
         return 0;
 }
 
diff --git a/units/systemd-logind.service.in b/units/systemd-logind.service.in
index 24f5ddaa17..042ea75d7a 100644
--- a/units/systemd-logind.service.in
+++ b/units/systemd-logind.service.in
@@ -58,7 +58,6 @@ StateDirectory=systemd/linger
 SystemCallArchitectures=native
 SystemCallErrorNumber=EPERM
 SystemCallFilter=@system-service
-Type=notify-reload
 {{SERVICE_WATCHDOG}}
 
 # Increase the default a bit in order to allow many simultaneous logins since
-- 
2.39.1

